lib/updatehub/updatehub.c

```diff
@@ -703,9 +705,18 @@ enum updatehub_response updatehub_probe(void)
 			goto cleanup;
 		}
 
+		sha256size = strlen(
+			metadata_any_boards.objects[1].objects.sha256sum);
+
+		if (sha256size != TC_SHA256_BLOCK_SIZE) {
```

Comment:
So, TC_SHA256_BLOCK_SIZE is completely unrelated to the size you are expecting here. I'd suggest creating a new define that is twice the hash size. The block size is an internal feature of the SHA256 algorithm, and it is only coincidence that it is twice the size of the hash output. I wouldn't expect there to be any uses of TC_SHA256_BLOCK_SIZE anywhere in this code.

Reply:
Created SHA256_HEX_DIGEST_SIZE to allow 64 byte binary stream be accommodated as hex string.